code/datums/spells/infinite_guns.dm

```diff
@@ -45,11 +45,119 @@
+	flags = NOBLUDGEON | DROPDEL | ABSTRACT
+
+/obj/item/gun/projectile/shotgun/boltaction/enchanted/arcane_barrage/fireball/customised_abstract_text(mob/living/carbon/owner)
+	. = ..()
+	return "<span class='warning'>[owner.p_their(TRUE)] [owner.l_hand == src ? "left hand" : "right hand"] is burning in magic fire.</span>"
 
 /obj/item/ammo_box/magazine/internal/boltaction/enchanted/arcane_barrage/fireball
 	ammo_type = /obj/item/ammo_casing/magic/arcane_barrage/fireball
 
 /obj/item/ammo_casing/magic/arcane_barrage/fireball
 	projectile_type = /obj/item/projectile/magic/fireball
 	muzzle_flash_effect = null
+
+/obj/item/gun/projectile/shotgun/boltaction/enchanted/arcane_barrage/fireball/linked
+	var/obj/effect/proc_holder/spell/infinite_guns/fireball/basic/source
+
+/obj/item/gun/projectile/shotgun/boltaction/enchanted/arcane_barrage/fireball/linked/Initialize(mapload, new_source)
+	. = ..()
+	source = new_source
+
+/obj/item/gun/projectile/shotgun/boltaction/enchanted/arcane_barrage/fireball/linked/Destroy()
+	source = null
+	return ..()
+
+/obj/effect/proc_holder/spell/infinite_guns/fireball/basic
+	name = "Fireball"
+	desc = "This spell fires a fireball at a target and does not require wizard garb."
+	base_cooldown = 6 SECONDS
+	clothes_req = FALSE
+	invocation = "ONI SOMA"
+	invocation_type = ""
+	var/invocation_type_on_shoot = "shout"
+	cooldown_min = 2 SECONDS
+	gun_type = /obj/item/gun/projectile/shotgun/boltaction/enchanted/arcane_barrage/fireball/linked/basic
+	action_icon_state = "fireball0"
+
+/obj/effect/proc_holder/spell/infinite_guns/fireball/basic/cast(list/targets, mob/user)
+	revert_cast() // to disable cooldown
+
+	var/mob/living/carbon/C = targets[1]
+	if(C.l_hand?.type == gun_type || C.r_hand?.type == gun_type)
+		remove_weapon(any_hand = TRUE)
+		return FALSE
+
+	if(!C.drop_item())
+		to_chat(user, "[user.get_active_hand()] is stuck to your hand, you cannot invoke a fireball over it!")
+		return FALSE
+
+	var/obj/item/gun/projectile/shotgun/boltaction/enchanted/GUN = new gun_type(FALSE, src)
+	C.put_in_hands(GUN)
+	RegisterSignal(C, COMSIG_MOB_WILLINGLY_DROP, PROC_REF(remove_weapon), override = TRUE)
+	C.changeNext_click(0) // allows to get fireball and instantly shoot it
+
+/obj/effect/proc_holder/spell/infinite_guns/fireball/basic/proc/remove_weapon(any_hand = FALSE)
+	SIGNAL_HANDLER
+	var/mob/owner = action.owner
+	var/obj/item/active_hand_item = owner.get_active_hand()
+	if(!any_hand && !(active_hand_item?.type ==  gun_type))
+		return
+
+	if(owner.l_hand?.type == gun_type)
+		qdel(owner.l_hand)
+		owner.update_inv_l_hand()
+		return
+
+	if(owner.r_hand?.type == gun_type)
+		qdel(owner.r_hand)
+		owner.update_inv_r_hand()
+
+/obj/effect/proc_holder/spell/infinite_guns/fireball/basic/proc/yell_on_shoot(mob/user)
+	invocation_type = invocation_type_on_shoot
+	invocation(user)
+	invocation_type = initial(invocation_type)
+
+/obj/item/gun/projectile/shotgun/boltaction/enchanted/arcane_barrage/fireball/linked/basic
```

Comment:
i am surprised its not a helicopter too 🤣

Reply:
that will be next fireball change for sure

Reply:
probably gonna move it to its own class though in future.. if that PR will not be closed because we dont want it